tests/components/blebox/test_config_flow.py

```diff
@@ -190,3 +192,113 @@ async def test_async_remove_entry(hass, valid_feature_mock):
 
     assert hass.config_entries.async_entries() == []
     assert config.state is config_entries.ConfigEntryState.NOT_LOADED
+
+
+async def test_flow_with_zeroconf(hass):
+    """Test setup from zeroconf discovery."""
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            host="172.100.123.4",
+            addresses=["172.100.123.4"],
+            port=80,
+            hostname="bbx-bbtest123456.local.",
+            type="_bbxsrv._tcp.local.",
+            name="bbx-bbtest123456._bbxsrv._tcp.local.",
+            properties={"_raw": {}},
+        ),
+    )
+
+    assert result["type"] == FlowResultType.FORM
+
+    with patch("homeassistant.components.blebox.async_setup_entry", return_value=True):
+        result2 = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "create_entry"
+    assert result2["data"] == {"host": "172.100.123.4", "port": 80}
+
+
+async def test_flow_with_zeroconf_when_already_in_progress(hass):
+    """Test behaviour if flow for device exists."""
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            host="172.100.123.4",
+            addresses=["172.100.123.4"],
+            port=80,
+            hostname="bbx-bbtest123456.local.",
+            type="_bbxsrv._tcp.local.",
+            name="bbx-bbtest123456._bbxsrv._tcp.local.",
+            properties={"_raw": {}},
+        ),
+    )
+
+    assert result["type"] == FlowResultType.FORM
```

Comment:
Please complete the config flow all the way to an ABORT or CREATE_ENTRY result

Reply:
Test's contract changed to check new config, against the already configured device.